packages/mountable/src/service-worker.ts

```
@@ -22,5 +33,59 @@ self.addEventListener("activate", (event: ExtendableEvent) => {
   event.waitUntil(self.clients.claim());
 });
 
-// Export to make this a module
+// Handle messages from the main thread
+self.addEventListener("message", (event: ExtendableMessageEvent) => {
+  const message = event.data as SWIncomingMessage;
+
+  if (message.type === SW_MESSAGE_TYPES.REGISTER_APP) {
+    const { cluster, project, appId, lastUpdatedTime, files } = message.payload;
+
+    try {
+      const response = registerApp(
+        appCaches,
+        cluster,
+        project,
+        appId,
+        lastUpdatedTime,
+        files,
+      );
+
+      // Send response back to the main thread
+      event.ports[0]?.postMessage(response);
```

When can this be undefined? I think maybe it should throw and not silently fail when its the case.

---

Seing that the same function is called several times I think its worth creating a `safePostMessage` or something that checks that `ports[0]` is defined before `postMessage` or throws if not. And; when is there more than one port? Should we also account for this?